app/auth/oauth_provider_app.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import requests
+import json
+
+PROVIDER_KINDS = {
+    "JUPYTERHUB": "jupyterhub",
+    "GITLAB": "gitlab",
+    "KEYCLOAK": "keycloak",
+}
+
+
+class OAuthProviderApp:
+    """A simple class combining some information about the oauth provider and the
+    application registered with the provider."""
+
+    def __init__(
+        self,
+        kind=None,
+        base_url=None,
+        client_id=None,
+        client_secret=None,
+        authorization_endpoint=None,
+        token_endpoint=None,
+    ):
+        self.kind = kind
+        self.base_url = base_url
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.authorization_endpoint = authorization_endpoint
+        self.token_endpoint = token_endpoint
+
+    def to_json(self):
+
+        SERIALIZER_ATTRIBUTES = [
```

Comment:
Make it global if it's constant or lowercase if it's function variable.

Reply:
changed to lowercase